tests/test_time_series/test_forecasting/test_window_features.py

```diff
@@ -41,187 +35,541 @@ def test_error_when_non_permitted_param_periods(_periods):
+            np.nan,
+            99.02,
+            102.19,
+            102.98,
+            100.24,
+            96.62,
         ],
-        "module_temp_window_2_freq_45min": [
-            np.nan, np.nan, np.nan, np.nan, 99.02, 102.19, 102.98, 100.24, 96.62
+        "irradiation_window_2_sum": [
+            np.nan,
+            np.nan,
+            np.nan,
+            np.nan,
+            1.3,
+            1.44,
+            1.41,
+            1.18,
+            0.91,
         ],
-        "irradiation_window_2_freq_45min": [
-            np.nan, np.nan, np.nan, np.nan, 1.3, 1.44, 1.41, 1.18, 0.91]
-        ,
     }
 
-    expected_results_df = pd.DataFrame(data=expected_results, index=date_time)
+    expected_results_df = pd.DataFrame(data=expected_results, index=_date_time)
 
-    transformer = WindowFeatures(window=2, function=np.sum, freq="45min")
-    transformer.fit(df_time)
-    df_tr = transformer.transform(df_time)
+    # Case 1: automatically select variables
+    transformer = WindowFeatures(window=2, functions=["sum"], freq="45min")
+    df_tr = transformer.fit_transform(df_time)
 
     assert df_tr.head(9).round(3).equals(expected_results_df)
 
-    # when drop_original is true
+    # Case 2: when drop_original is true
     transformer = WindowFeatures(
-        window=2, function=np.sum, freq="45min", drop_original=True
+        window=2, functions=["sum"], freq="45min", drop_original=True
     )
-    transformer.fit(df_time)
-    df_tr = transformer.transform(df_time)
-
-    assert df_tr.head(9).round(3).equals(
-        expected_results_df.drop(["ambient_temp", "module_temp", "irradiation"], axis=1)
+    df_tr = transformer.fit_transform(df_time)
+
+    assert (
+        df_tr.head(9)
+        .round(3)
+        .equals(
+            expected_results_df.drop(
+                ["ambient_temp", "module_temp", "irradiation"], axis=1
+            )
+        )
     )
 
-    # select variables
+    # Case 3: user indicates multiple variables
     transformer = WindowFeatures(
         variables=["ambient_temp", "irradiation"],
         window=2,
-        function=np.sum,
-        freq="45min"
+        functions=["sum"],
+        freq="45min",
     )
-    transformer.fit(df_time)
-    df_tr = transformer.transform(df_time)
+    df_tr = transformer.fit_transform(df_time)
+
+    assert (
+        df_tr.head(9)
+        .round(3)
+        .equals(expected_results_df.drop(["module_temp_window_2_sum"], axis=1))
+    )
+
+    # Case 4: user indicates 1 variable
+    transformer = WindowFeatures(
+        variables=["irradiation"], window=2, functions=["sum"], freq="45min"
+    )
+    df_tr = transformer.fit_transform(df_time)
+
+    assert (
+        df_tr.head(9)
+        .round(3)
+        .equals(
+            expected_results_df.drop(
+                ["ambient_temp_window_2_sum", "module_temp_window_2_sum"], axis=1
+            )
+        )
+    )
+
+
+# TODO: complete the values for this test
+def test_multiple_windows(df_time):
```

Comment:
we need to adjust the numbers of this test, wherever it says xxx to the expected results